### PR TITLE
[fix]: [SRM-15677]: mark analysisStateMachine to a different state whose slo has been deleted and terminate the orchestrator

### PR DESCRIPTION
**Pull request by n00bitax**
## Describe your changes
mark analysisStateMachine to a different state whose slo has been deleted and terminate the orchestrator 
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54240)

### CODE REVIEW

**Comment by n00bitax** [comment]
trigger uts

**Comment by n00bitax** [comment]
trigger uts

**Comment by n00bitax** [comment]
trigger uts

**Comment by n00bitax** [comment]
trigger messagecheck

**Comment by n00bitax** [comment]
trigger ut6

**Comment by n00bitax** [comment]
trigger ut7

**Comment by n00bitax** [comment]
trigger ut6,ut7

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/statemachine/services/api/CompositeSLOMetricAnalysisStateExecutor.java [hunk 1]
```diff
@@ -71,10 +72,16 @@ public AnalysisState execute(CompositeSLOMetricAnalysisState analysisState) {
       }
       analysisState.setStatus(AnalysisStatus.SUCCESS);
     } catch (Exception exception) {
-      log.warn(String.format("Composite SLO Execute for sloId: {} failed with:",
-                   analysisState.getInputs().getVerificationTaskId()),
-          exception);
-      analysisState.setStatus(AnalysisStatus.RETRY);
+      if (compositeServiceLevelObjective == null) {
+        analysisState.setStatus(AnalysisStatus.MARKED_FOR_DELETION);
+        log.info("The sloId {} has already been deleted, marking current analysisState to be cleaned up : {}", sloId,
```
Comment (Varsha-lalwani) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/statemachine/services/api/CompositeSLOMetricAnalysisStateExecutor.java:77]: composite slo with id*

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/statemachine/services/impl/OrchestrationServiceImpl.java [hunk 2]
```diff
@@ -231,6 +231,7 @@ private void orchestrateAtRunningState(AnalysisOrchestrator orchestrator) {
           try {
             context = new AnalysisStateMachineContext(currentlyExecutingStateMachine);
             stateMachineStatus = stateMachineService.executeStateMachine(currentlyExecutingStateMachine);
+            System.out.println("Status |||" + stateMachineStatus);
```
Comment (Varsha-lalwani) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/statemachine/services/impl/OrchestrationServiceImpl.java:234]: remove

File: srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/statemachine/services/impl/OrchestrationServiceImplTest.java [hunk 3]
```diff
@@ -668,6 +672,88 @@ public void testQueueAnalysis_forSLIVerificationTask() {
         .isEqualTo(AnalysisState.StateType.SLI_METRIC_ANALYSIS);
   }
 
+  @Test
+  @Owner(developers = ANSUMAN)
+  @Category(UnitTests.class)
+  public void testExecuteAfterSLODeletion() {
```
Comment (Varsha-lalwani) [inline srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/statemachine/services/impl/OrchestrationServiceImplTest.java:678]: AfterCompositeSLODeletion* would be better to tell what test is doing

File: srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/statemachine/services/impl/OrchestrationServiceImplTest.java [hunk 4]
```diff
@@ -668,6 +672,88 @@ public void testQueueAnalysis_forSLIVerificationTask() {
...
+    serviceLevelObjectiveV2Service.delete(
+        builderFactory.getProjectParams(), simpleServiceLevelObjective1.getIdentifier());
+    orchestrate(verificationTaskId);
+    orchestrate(verificationTaskId);
+    AnalysisOrchestrator orchestrator = hPersistence.createQuery(AnalysisOrchestrator.class)
+                                            .filter(AnalysisOrchestratorKeys.verificationTaskId, verificationTaskId)
+                                            .get();
+    assertThat(orchestrator).isNotNull();
+    assertThat(orchestrator.getUuid()).isNotNull();
+    assertThat(orchestrator.getStatus().name()).isEqualTo(AnalysisStatus.TERMINATED.name());
```
Comment (Varsha-lalwani) [inline srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/statemachine/services/impl/OrchestrationServiceImplTest.java:753]: can you also add validaton for state machine status

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/statemachine/services/api/CompositeSLOMetricAnalysisStateExecutor.java [hunk 5]
```diff
@@ -43,12 +43,13 @@ public class CompositeSLOMetricAnalysisStateExecutor extends AnalysisStateExecut
...
   public AnalysisState execute(CompositeSLOMetricAnalysisState analysisState) {
+    CompositeServiceLevelObjective compositeServiceLevelObjective = null;
+    String sloId = null;
     try {
       String verificationTaskId = analysisState.getInputs().getVerificationTaskId();
       // here startTime will be the prv Data endTime and endTime will be the current time.
-      String sloId = verificationTaskService.getCompositeSLOId(verificationTaskId);
-      CompositeServiceLevelObjective compositeServiceLevelObjective =
-          (CompositeServiceLevelObjective) serviceLevelObjectiveV2Service.get(sloId);
+      sloId = verificationTaskService.getCompositeSLOId(verificationTaskId);
```
Comment (Varsha-lalwani) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/statemachine/services/api/CompositeSLOMetricAnalysisStateExecutor.java:51]: what if verificationTask is deleted and sloId is null?

**Review by Varsha-lalwani — CHANGES_REQUESTED** [review]
Please make the changes

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/statemachine/services/api/CompositeSLOMetricAnalysisStateExecutor.java (hunk 5 above)
Comment (n00bitax) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/statemachine/services/api/CompositeSLOMetricAnalysisStateExecutor.java:51]: in that case, also, we will not have the compositeServiceLevelObjective object also right, as we can only fetch it if all of the above are not null.

**Comment by n00bitax** [comment]
trigger uts

**Comment by n00bitax** [comment]
trigger uts

**Review by Varsha-lalwani — APPROVED** [review]

**Pull request merged**